scripts/easy_train.py

```diff
@@ -0,0 +1,1844 @@
+        for gpu_id in gpu_ids:
+            for j in range(args.runs_per_gpu):
+                run_id = gpu_id*args.runs_per_gpu+j
+
+                training_runs.append(TrainingRun(
+                    gpu_id=gpu_id,
+                    run_id=run_id,
+                    nnue_pytorch_directory=nnue_pytorch_directory,
+                    training_dataset=args.training_dataset,
+                    validation_dataset=args.validation_dataset,
+                    num_data_loader_threads=args.num_workers,
+                    num_pytorch_threads=args.threads,
+                    num_epochs=args.max_epoch,
+                    batch_size=args.batch_size,
+                    random_fen_skipping=args.random_fen_skipping,
+                    smart_fen_skipping=args.smart_fen_skipping,
+                    wld_fen_skipping=args.wld_fen_skipping,
+                    features=args.features,
+                    lr=args.lr,
+                    gamma=args.gamma,
+                    network_save_period=args.network_save_period,
+                    save_last_network=args.save_last_network,
+                    seed=args.seed + run_id,
+                    start_from_model=start_model,
+                    root_dir=os.path.join(experiment_directory, 'training', f'run_{run_id}'),
+                    epoch_size=args.epoch_size,
+                    validation_size=args.validation_size,
+                    resume_training=args.resume_training,
+                    additional_args=[arg for arg in args.additional_training_args or []]
+                ))
+        LOGGER.info(f'Doing network training on gpus {gpu_ids}. {len(training_runs)} runs in total.')
+    else:
+        LOGGER.info('Not training networks.')
+
+    network_testing = NetworkTesting(
+        nnue_pytorch_directory=nnue_pytorch_directory,
+        root_dir=os.path.join(experiment_directory, 'training'),
+        ordo_exe=None if args.do_approximate_ordo else make_ordo_executable_path(ordo_directory),
+        c_chess_cli_exe=make_c_ches_cli_executable_path(c_chess_cli_directory),
+        stockfish_base_exe=make_stockfish_executable_path(stockfish_base_directory),
+        stockfish_test_exe=make_stockfish_executable_path(stockfish_test_directory),
+        features=args.features,
+        num_parallel_games=args.network_testing_threads,
+        explore_factor=args.network_testing_explore_factor,
+        book_file_path=args.network_testing_book,
+        time_per_game=args.network_testing_time_per_game,
+        time_increment_per_move=args.network_testing_time_increment_per_move,
+        nodes_per_move=args.network_testing_nodes_per_move,
+        hash=args.network_testing_hash_mb,
+        games_per_round=args.network_testing_games_per_round,
+        active=do_network_testing,
+        additional_args=[arg for arg in args.additional_testing_args or []]
+    )
+
+    #TODO it would be nice to be able to set a time limit at the cmdline and stop after e.g. 20h
```

Comment:
This would be a real nice feature for me, i.e. a clean shutdown (like 'quit') after a specified time.

Reply:
ok. Added experimentally `--auto_exit_timeout` and `--auto_exit_timeout_on_training_finished`. Can't test right now because doesn't work on windows (needs SIGALRM).

Reply:
one thing I worry about is that it may get swallowed by some `except` if thrown in a bad moment. We'd need to explicitely catch every exception to make this work 100% of the time.

Reply:
alternatively, reraise every 1 second?